CAP/gap/TerminalCategory.gi

```diff
@@ -4,55 +4,414 @@
+            """;
+            
+        end;
+    
+    ##
+    object_constructor := function( cat, string )
+        
+        return ObjectifyObjectForCAPWithAttributes(
+                       rec( ), cat,
+                       String, string );
+        
+    end;
+    
+    object_datum := { cat, object } -> String( object );
+    
+    morphism_constructor := function( cat, source, string, range )
+        
+        return ObjectifyMorphismWithSourceAndRangeForCAPWithAttributes(
+                       rec( ), cat,
+                       source,
+                       range,
+                       String, string );
+        
+    end;
+    
+    morphism_datum := { cat, morphism } -> fail;
+    
+    ## prevent skeletality and strictness
+    properties := Set( List( CAP_INTERNAL_CATEGORICAL_PROPERTIES_LIST, a -> a[1] ) );
+    excluded_properties := Filtered( properties, p -> IsInt( PositionSublist( p, "Strict" ) ) );
+    Add( excluded_properties, "IsSkeletalCategory" );
+    
+    T := CAP_INTERNAL_CONSTRUCTOR_FOR_TERMINAL_CATEGORY( rec(
+                 name := name,
+                 category_filter := category_filter,
+                 category_object_filter := category_object_filter,
+                 category_morphism_filter := category_morphism_filter,
+                 excluded_properties := excluded_properties,
+                 create_func_object := create_func_object,
+                 create_func_morphism := create_func_morphism,
+                 create_func_morphism_or_fail := create_func_morphism,
+                 object_constructor := object_constructor,
+                 object_datum := object_datum,
+                 morphism_constructor := morphism_constructor,
+                 morphism_datum := morphism_datum
+                 ) );
+    
+    ##
+    AddIsWellDefinedForObjects( T,
+      function( T, object )
+        
+        return IsString( String( object ) );
+        
+    end );
+    
+    ##
+    AddIsWellDefinedForMorphisms( T,
+      function( T, morphism )
+        
+        return IsString( String( morphism ) );
+        
+    end );
+    
+    ##
+    AddIsEqualForObjects( T,
+      function( T, object1, object2 )
+        
+        return String( object1 ) = String( object2 );
+        
+    end );
+    
+    ##
+    AddIsEqualForMorphisms( T,
+      function( T, morphism1, morphism2 )
+        
+        ## equality of source and target is part of the specification of the input and checked by the pre-function
+        return String( morphism1 ) = String( morphism2 );
+        
+    end );
+    
+    Finalize( T );
+    
+    return T;
+    
+end );
 
-InstallValue( CAP_INTERNAL_TERMINAL_CATEGORY_AS_CAT_OBJECT,
-              
-              AsCatObject( CAP_INTERNAL_TERMINAL_CATEGORY ) );
+##################################
+#
+# Attribute methods
+#
+##################################
 
 ##
 InstallMethod( UniqueObject,
                [ IsCapCategory and IsTerminalCategory ],
```

Comment:
This has to be specialized to `IsTerminalCategoryWithSingleObject` or `IsTerminalCategoryWithUniqueObject`.

Reply:
I also changed the other types in `TerminalCategory.gd` for a terminal category with a single object to reflect this. I think this makes more sense. Is this fine?

Reply:
Yes, that was implied, thanks!